### PR TITLE
Do not silently ignore SMTP errors

### PR DESCRIPTION
**Pull request by pferreir**
Since we don't need anymore to retry transactions (as was the case with ZODB), we can just send any e-mails before the commit takes place.

### CODE REVIEW

File: indico/legacy/webinterface/rh/base.py [hunk 1]
```diff
@@ -551,13 +551,13 @@ def _do_process(self, profile):
...
 
     def _process_success(self):
-        Logger.get('requestHandler').info('Request successful')
-        # request is succesfull, now, doing tasks that must be done only once
+        logger = Logger.get('requestHandler')
+        logger.info('Request successful')
+        # request is succesfull, let's get rid of temporary files
         try:
-            GenericMailer.flushQueue(True)  # send emails
             self._deleteTempFiles()
```
Comment (ThiefMaster) [inline indico/legacy/webinterface/rh/base.py]: This is actually obsolete and can go away - `self._tempFilesToDelete` is always an empty list in both RH and ServiceBase

File: indico/legacy/webinterface/rh/base.py [hunk 2]
```diff
@@ -551,13 +551,13 @@ def _do_process(self, profile):
         return profile_name, res
 
     def _process_success(self):
-        Logger.get('requestHandler').info('Request successful')
-        # request is succesfull, now, doing tasks that must be done only once
+        logger = Logger.get('requestHandler')
+        logger.info('Request successful')
+        # request is succesfull, let's get rid of temporary files
```
Comment (ThiefMaster) [inline indico/legacy/webinterface/rh/base.py]: successful

File: indico/legacy/webinterface/rh/base.py [hunk 3]
```diff
@@ -587,6 +587,15 @@ def process(self, params):
                 self._check_auth(params)
                 profile_name, res = self._do_process(profile)
                 signals.after_process.send()
+
+                if self.commit:
+                    # ensure we fail early (before sending out e-mails)
+                    # in case there are DB constraint violations, etc...
+                    db.session.flush()
```
Comment (ThiefMaster) [inline indico/legacy/webinterface/rh/base.py]: You could also call `db.enforce_constraints()` instead - this already flushes but also ensures that e.g. inconsistent-timetable errors are triggered

Comment (pferreir) [inline indico/legacy/webinterface/rh/base.py]: Oh cool.

**Pull request merged**